### DISCUSSION TITLE
Why should I specify DPI when creating a Bitmap Image and how should I do it

### DISCUSSION THREAD

**Discussion by martinrhan** [AvaloniaUI/Avalonia#10551] [Q&A]
By what I know DPI is only appropriate in the context of describing quality of output device, in which describes how dense does pixels distribute on the plane of monitor or printed paper. But how does it make sense that I need to pass in an argument specifying DPI when calling constructor of WriteableBitmap.

I tried looking at property of an image file in my computer, it only has information of size, width * height in unit of pixel, and how many bit does each pixel take. That suggests image file itself don't need DPI information to be displayed on screen.

**Comment by kekekeks** [comment]
IIRC it's about logical size and pixel density. I. e. a 200x200 pixel bitmap with 192 DPI is supposed to have 100x100 logical size.

**Reply by martinrhan** [reply]
What do you mean by logical size?

**Reply by kekekeks** [reply]
The size reported by `Bitmap.Size` and later used by layout and tiling.
